Creation of Lib service

ravvnen opened the pull request.

prasadtalasila commented:
@ravvnen 
A few comments:
* move test files from servers/lib/src to servers/lib/test
* both unit and integration tests are required

More comments to follow later today.

SyntaxXeror commented on servers/lib/.eslintrc.js:
```diff
@@ -0,0 +1,25 @@
+  },
+  ignorePatterns: ['.eslintrc.js'],
+  rules: {
+    '@typescript-eslint/interface-name-prefix': 'off',
+    '@typescript-eslint/explicit-function-return-type': 'off',
+    '@typescript-eslint/explicit-module-boundary-types': 'off',
+    '@typescript-eslint/no-explicit-any': 'off',
```
By turning these rules off it's like forcing eslint to not format for TypeScript and allow javaScript, which would be kind of redundant to even use TypeScript then.
You are specifically turning off return type declaration in line 21. Why use TypeScript then?

ravvnen replied:
I had not noticed that, ive made the change, so that the are all set to 'on', thanks!

SyntaxXeror replied:
We just had a meeting with Prasad, he suggested that our group and you plan a meeting to look at getting common config and script files, so maybe let this be standby for now until then.

ravvnen replied:
Sounds like a good idea - Do you have time monday morning/noon?

prasadtalasila commented on servers/lib/test/files.e2e-spec.ts:
```diff
@@ -0,0 +1,61 @@
+    filesService = moduleFixture.get<FilesService>(FilesService);
+  });
+
+    it('should return an array of file names for a given path', async() => {
```
This is a unit test. Please rename the file to indicate that it's an integration test but not end2end.
An end2end test requires running the complete app with "yarn start:test" and then running tests using graphql queries sent from the test code to the live microservice.

prasadtalasila replied:
Doing this end2end test would take the integration test coverage to near 100%.

ravvnen replied:
I have just pushed the update of app.e2e-spec.ts

prasadtalasila commented on servers/lib/test/files.service.spec.ts:
```diff
@@ -0,0 +1,151 @@
+    it("should throw an error if response from GitLab API is invalid", async () => {
+      jest
+        .spyOn(ApolloClient.prototype, "query")
+        .mockImplementation(() => ApolloClient as any);
```
is it possible to fix this?

ravvnen replied:
change any to a more explicit type

ravvnen commented on servers/lib/README.md:
```diff
@@ -0,0 +1,30 @@
+
+This lib-microservice takes a query from the user that asks for content within a given specified directory. This microservices handles request by fetching and returning the file-names and folders within that respective directory.
+
+# Gitlab setup
```
double hash

ravvnen commented on servers/lib/src/main.ts:
```diff
@@ -0,0 +1,13 @@
+  const configService = app.get(ConfigService);
+  const port = configService.get<number>("PORT");
+  const mode = configService.get<string>("MODE");
+  console.log("running mode", mode);
```
remove console

ravvnen commented:
Hello @prasadtalasila, i have just fixed and implemented the following, as we discussed in the meeting:
deleted unncessesary scripts in my scripts file
update README to include development instructions
hardcode in return of "invalid query" in filesservice
fixed test cases to match returning of invalid query, and not throwing error

So if you want, it should be able to merge now. 
I will look into the github action and yaml file now and thereafter the codeclimate issues, if you are content with everything else 😄

prasadtalasila commented:
@ravvnen the Github Actions are blocking due to delay in closing an orphan process and ultimately failing. Can you please push a branch in your fork and check the results of the Github Actions? If it's successful there, you can update the PR branch.

ravvnen commented:
@prasadtalasila i have created a development branch where i will try to run my github action. 

i am not quite understanding why my 'path' is failing, since its a variable i am passing in each test case, but i am looking into it.

ravvnen commented:
Hello @prasadtalasila 😄 
I have now:
Updated package.json to maintain consistency with the script, aswell as check dependencies.
Created server.yml, and set it up so that my tests are able to run and succeeed, which you should be able to view on my "testing" branch: https://github.com/ravvnen/DTaaS/actions/runs/4557246823

The warning i am receiving from typescript, i am ignoring, since i am actually using the value later on. 



I will now:
Look into codeclimate analysis,
Making the codecov uploader function

prasadtalasila commented:
Completed in PR #9
